tests/unit/src/main/scala/tests/BaseAmmoniteSuite.scala

```diff
@@ -0,0 +1,238 @@
+package tests
+
+import org.eclipse.lsp4j.MessageActionItem
+import scala.meta.internal.metals.Messages
+
+abstract class BaseAmmoniteSuite(scalaVersion: String)
+    extends BaseLspSuite("ammonite") {
+
+  override def munitIgnore: Boolean =
+    !isValidScalaVersionForEnv(scalaVersion) ||
+      // temporary, remove once Ammonite is available for 2.12.11
+      scalaVersion == "2.12.11"
+
+  override def newServer(workspaceName: String): Unit = {
+    super.newServer(workspaceName)
+    server.client.showMessageRequestHandler = { params =>
+      if (params == Messages.ImportAmmoniteScript.params())
+        Some(new MessageActionItem(Messages.ImportAmmoniteScript.dismiss))
+      else
+        None
+    }
+  }
+
+  test("simple script") {
+    // single script with import $ivy-s
+    for {
+      _ <- server.initialize(
+        s"""
+           |/metals.json
+           |{
+           |  "a": {
+           |    "scalaVersion": "$scalaVersion"
+           |  }
+           |}
+           |/main.sc
+           | // scala $scalaVersion
+           |import $$ivy.`io.circe::circe-core:0.12.3`
+           |import $$ivy.`io.circe::circe-generic:0.12.3`
+           |import $$ivy.`io.circe::circe-parser:0.12.3`
+           |import io.circe._, io.circe.generic.auto._, io.circe.parser._, io.circe.syntax._
+           |
+           |sealed trait Foo
+           |case class Bar(xs: Vector[String]) extends Foo
+           |case class Qux(i: Int, d: Option[Double]) extends Foo
+           |
+           |val foo: Foo = Qux(13, Some(14.0))
+           |
+           |val json = foo.asJson.noSpaces
+           |
+           |val decodedFoo = decode[Foo](json)
+           |""".stripMargin
+      )
+      _ <- server.didOpen("main.sc")
+      _ <- server.didSave("main.sc")(identity)
+      _ <- server.executeCommand("ammonite-start")
+
+      // asJson
+      // via Ammonite-generated Semantic DB
+      _ <- server.assertDefinitionAtLocation(
+        "main.sc",
+        12,
+        18,
+        ".metals/readonly/io/circe/syntax/package.scala"
+      )
+
+      // noSpaces
+      // via Ammonite-generated Semantic DB
+      _ <- server.assertDefinitionAtLocation(
+        "main.sc",
+        12,
+        27,
+        ".metals/readonly/io/circe/Json.scala"
+      )
+
+      // Second 'noSpaces' in
+      //   final def noSpaces: String = Printer.noSpaces.print(this)
+      // via presentation compiler, using the Ammonite build target classpath
+      _ <- server.assertDefinitionAtLocation(
+        ".metals/readonly/io/circe/Json.scala",
+        122,
+        45,
+        ".metals/readonly/io/circe/Printer.scala"
+      )
+
+      // Foo
+      // via Ammonite-generated Semantic DB and indexing of Ammonite-generated source of $file dependencies
+      _ <- server.assertDefinitionAtLocation("main.sc", 14, 25, "main.sc", 6)
+
+      expectedHoverRes = """```scala
```

Comment:
Would be nice to do it the same way as in `HoverLspSuite`

Reply:
This can also be a separate test case I think. It could be in a method `checkHovers`